### ISSUE TITLE
[Blazor Hybrid] BlazorWebView ComponentType error blocks XAML Hot Reload updates 

### ISSUE THREAD

**Issue by nor0x** [dotnet/maui#20669]
### Description

just created a File -> New .NET MAUI Blazor Hybrid app in Visual Studio and started a Debug Build on Windows. XAML Hot-Reload is not working. There are multiple of these logs in the `Hot Reload Console`.

```
19:08 04.52 MauiApp24: Hot Reload session started
19:11 14.01 No changes were found.
19:11 24.95 No changes were found.
```

It seems that only XAML HR is not working for me, changes in .razor files are updated correctly.

### Steps to Reproduce

1. Create new MAUI Blazor Hybrid app
2. Debug on Windows or Android
3. Make XAML changes

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows and Android, I was *not* able test on other platforms

### Affected platform versions

- Windows 23H2
- Android

### Relevant log output

```shell
My Visual Studio version is:

Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview
Version 17.10.0 Preview 1.0
```

**Comment by drasticactions** [comment]
XAML Hot Reload is a Visual Studio feature and isn't something that can be addressed in this repo, you should file this to VSFeedback.

Also, those logs you posted are for C# Hot Reload, not XAML.

**Comment by drasticactions** [comment]
Also, checking the newest Visual Studio 17.9 and 17.10 Preview builds, it seems to work correctly for me on Android and Windows,

![image](https://github.com/dotnet/maui/assets/898335/e68e9dd6-c413-48e4-808c-ca342886f897)

If you see the Live Visual Tree, then XAML Hot Reload functions. If you can edit properties in Live Property Explorer, it functions. Since you mentioned Razor changes being applied, those occur through the XAML Hot Reload system, so the connection should be working. 

I assume that you're changing specific properties to BlazorWebView then and those are not applying. Trying to edit it, `ComponentType` doesn't seem to like to be changed or modified so if you try and edit code around it then it might protest and not apply changes to the rest. That's something that could be looked at within this repo.

So TL;DR, if you don't see things like Live Visual Tree or Live Property Explorer, or you get yellow bars or errors saying XAML Hot Reload can't start, that should go to VS since it can't be fixed here.

If you are trying to change specific things and those are not applying, then file it here and we can move or modify the issue as needed depending on where the fix has to go: The core control or the underlying system itself.

**Comment by nor0x** [comment]
thank you for the reply @drasticactions. Indeed there are some logs in the Xamarin Hot Reload panel:

```
[11:00:52]  (MauiApp24) .NET Project references .NET MAUI.
[11:00:52]  (MauiApp24) WinUI hot reload is in progress.
[11:01:06]  Hot Reload changes cannot be applied due to this error in the XAML:
  D:\Dev\maui\MauiApp24\MainPage.xaml(16,59): Nested types are not supported: Components.Routes.
```
which is also displayed in the XAML Editor. I can confirm that the Live Visual Tree is filled and changing properties via the Live Property Explorer works fine for me.

![image](https://github.com/dotnet/maui/assets/3210391/2ac1a8b5-4349-458e-bc44-297ed32524d9)

In my case nothing is hot-reloaded right now - I tried changing i.e. the `BackgroundColor `of the `ContentPage` or the outer `Border` or `Grid`

**Comment by nor0x** [comment]
thank you for the info and explanation. Interesting to understand what's going on here. For now I'm unblocked by removing the  `ComponentType` during runtime - but it's not nice though. I think this is something that can be handled in a better way by the tooling / framework.  Also I don't quite understand why it works for you in a new blank project 🤔

**Comment by QianaJiao** [comment]
Verified this issue with Visual Studio 17.10.0 Preview 2 (8.0.14/8.0.7), not repro on windows platform.

**Comment by nor0x** [comment]
just tried with the latest Visual Studio Preview and I can still reproduce this one.

1. Create new Maui Blazor Hybrid Project
2. Start Debug (Hot-Reload Session)
3. Wrap BlazorWebView inside Grid

![image](https://github.com/dotnet/maui/assets/3210391/101d533d-a738-44b0-9239-ee6f34ab86be)

In this case I'm not directly editing properties of the `BlazorWebView`, but wrap the control inside a layout or so - for me in this case this is 100% reproducible

**My Visual Studio Info**
Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview
Version 17.10.0 Preview 2.0

**Comment by QianaJiao** [comment]
Follow the steps above to revalidate, and the result is it repro.
It seems that whenever you edit MainPage.xaml in debug mode,  ComponentType="{x:Type local:Components.Routes} will report an error.

**Comment by nor0x** [comment]
is there anything happening regarding this issue @drasticactions ? really annoying limitation that slows us down at development because we cannot hot-reload anything in our application

**Comment by drasticactions** [comment]
Short answer is No, nothing specifically for this has changed as far as I'm aware.

Slightly longer answer is:

- The Live Property Explorer does work, so you could still use that to edit properties on the Blazor view, or other properties within it.
- C# Hot Reload and editing Razor views are unrelated to this issue, so if that doesn't work you would need to file another issue.
- You could work around it by putting a placeholder view in and editing your layout, then putting it back when you're done. 

@etvorun @mgoertz-msft Is there a way to mask this editor error so we can keep letting requests through?

**Comment by nor0x** [comment]
thanks for the reply, the hot-reload story is really not great right now

- Live Property Explorer is not an Option - I don't want to change properties on the `BlazorWebView` itself but for example change a XAML `Button` on the `ContentPage`
- Razor Hot-Reload is broken because of a different error `[Error] C:\dev\MauiRepros\MauiApp68.sln (line 1): error ENC1002: Cannot apply changes -- unexpected error: 'Unexpected value 'MauiApp.Home' of type 'Microsoft.CodeAnalysis.CSharp.Symbols.SourceNamedTypeSymbol''`
- a placeholder view might work in a basic app but I have elements in the `BlazorWebView` that trigger changes to XAML elements

**Comment by nor0x** [comment]
was able to find a workaround myself. I manually moved `Routes.razor` and `MainLayout.razor` to the root namespace of the app by adding `@namespace MauiApp2`. By doing this I was able to remove the nested type from `...ComponentType="{x:Type local:Components.Routes}" />` and just do `...ComponentType="{x:Type local:Routes}" />`


glad that this works now for me - still lame that this is broken in the default template of a MAUI Blazor Hybrid app

BEFORE:
![image](https://github.com/dotnet/maui/assets/3210391/3739dae3-7ca0-467a-b64d-3bef4fd446db)

AFTER:
![image](https://github.com/dotnet/maui/assets/3210391/8848b593-730e-4865-ab1f-48dcf25e9150)

**Comment by shushu789** [comment]
For me, I also encountered a similar problem, which mainly occurred on the Android platform. When modifying the razor code, an error would occur: An unexpected error occurred during hot reload, regardless of server or wasm mode. But no problem was found on winui and web

**Comment by mattleibow** [comment]
@QianaJiao is this still happening in the new .NET maui and/or VS?

@drasticactions what can we do to make this better? Can we avoid this by updating the templates?

@StephaneDelcroix  the "nested types are not supported" error is sounding like this is a xaml parser issue?

**Comment by nor0x** [comment]
> Hi [@nor0x](https://github.com/nor0x). We have added the "s/try-latest-version" label to this issue, which indicates that we'd like you to try and reproduce this issue on the latest available public version. This can happen because we think that this issue was fixed in a version that has just been released, or the information provided by you indicates that you might be working with an older version.
> 
> You can install the latest version by installing the latest Visual Studio (Preview) with the .NET MAUI workload installed. If the issue still persists, please let us know with any additional details and ideally a [reproduction project](https://github.com/dotnet/maui/blob/main/.github/repro.md) provided through a GitHub repository.
> 
> This issue will be closed automatically in 7 days if we do not hear back from you by then - please feel free to re-open it if you come back to this issue after that time.

same with .NET 9 - File -> New -> Blazor Maui Hybrid -> Edit XAML during debug
![Image](https://github.com/user-attachments/assets/96d42b26-3274-4960-80f9-8a7063861d2d)

**Comment by BretJohnson** [comment]
The cause of this issue is that Hot Reload in Visual Studio currently doesn't support qualified name syntax, so for this `ComponentType="{x:Type local:Components.Routes}"` Hot Reload gives this error `XHR0010: Nested types are not supported: Components.Routes.`

The workaround is to update the XAML so that the qualified name syntax isn't used, by adding a new namespace:
`xmlns:components="clr-namespace:MauiBlazorTest.Components"` and then updating the binding to use it instead: `ComponentType="{x:Type components:Routes}"`.

@mattleibow - it'd really be best for us to update the template to do this by default, avoiding the problem altogether. I can create a PR. Do you think this is something we may be able to get into a .NET9 SR? Not only does the default template not work with MAUI XAML Hot Reload, but any other XAML pages users create by copying the default one won't work either today.

**Comment by mattleibow** [comment]
We can update the templates, and we can get this in for the SR that will be merged into VS.

**Comment by jaliyaudagedara** [comment]
@BretJohnson, 

I started seeing the same the error:
![Image](https://github.com/user-attachments/assets/e7adc082-5b57-4be9-bba2-0cb5a837ee87)

And followed your suggestion.
```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:local="clr-namespace:MauiApp2"
             xmlns:components="clr-namespace:MauiApp2.Components"
             x:Class="MauiApp2.MainPage">

    <BlazorWebView x:Name="blazorWebView" HostPage="wwwroot/index.html">
        <BlazorWebView.RootComponents>
            <RootComponent Selector="#app" ComponentType="{x:Type components:Routes}" />
        </BlazorWebView.RootComponents>
    </BlazorWebView>

</ContentPage>
```
Still the same:
![Image](https://github.com/user-attachments/assets/c14564e7-c56d-488d-8af5-f02fb7597119)

`Visual Studio 17.14.0 Preview 2.0`
